### ISSUE TITLE
superfluous_disable_command not triggering for custom rule

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#4754]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`superfluous_disable_command` is not triggering for custom rules. I don't know if this has always been the behavior or if it's a recent regression.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint file.swift           
Linting Swift files at paths file.swift
Linting 'file.swift' (1/1)
/Users/marcelo.fabri/SwiftLint/test/file.swift:1:5: warning: forbidden Violation: This is forbidden (forbidden)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.50.3
* Installation method used (Homebrew, CocoaPods, building from source, etc)? brew
* Paste your configuration file:

```yml
custom_rules:
  forbidden:
    regex: 'FORBIDDEN'
    message: "This is forbidden"
    severity: warning
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)? 14.0.1
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
let FORBIDDEN = 1
// swiftlint:disable:next forbidden
let ALLOWED = 2
```

**Comment by marcelofabri** [comment]
reopening because the PR that fixed it got reverted

**Comment by mildm8nnered** [comment]
So I picked this up again the other day, and I think I can see where we were going wrong before ... should have a PR soon

**Comment by mildm8nnered** [comment]
Resolved by #5670
